crates/bevy_transform/src/components/transform.rs

```diff
@@ -383,6 +383,20 @@ impl Transform {
         self.rotate(rotation);
     }
 
+    /// Flips the scale of this [`Transform`] based on which axes are not zero in the given `Dir3`.
+    #[inline]
+    pub fn flip(&mut self, dir: Dir3) {
```

Comment:
What motivates this API?

Specifically why should we use a vector of zero/nonzero values to specify the flips. (Does this vector have some geometric meaning?)

And second, why specifically a Dir3, the implementation doesn't require the normalisation, do we expect the user to have a Dir3 already available when calling this? It would probably be easier to convert a Dir3 to a Vec3 instead of vice-versa.

Reply:
Probably because providing a non-uniform input would produce surprising results with the current implementation.